server/src/test/java/com/scalar/db/server/DistributedTransactionServiceTest.java

```diff
@@ -836,6 +839,44 @@ public void abort_ManagerThrowsTransactionException_ShouldThrowInternalError()
     assertThat(exceptionCaptor.getValue().getStatus().getCode()).isEqualTo(Code.INTERNAL);
   }
 
+  @Test
+  public void
+      commit_TransactionThrowsCommitUnsatisfiedConditionException_ShouldRespondWithUnsatisfiedConditionError()
+          throws TransactionException {
```

Comment:
Nit. We should add the same test for `TwoPhaseCommitTransactionService`?

Reply:
Good catch, thanks. I added unit test for it in TwoPhaseCommitTransactionService and on the client side too.